File: AssetManager/Data/DataComms/AssetManager/FTP/FTPComms.vb
Imports System.Net

Public Class FtpComms

#Region "Fields"

    Private Const EncFTPUserPass As String = "BzPOHPXLdGu9CxaHTAEUCXY4Oa5EVM2B/G7O9En28LQ="
    Private Const strFTPUser As String = "asset_manager"
    Private FTPcreds As NetworkCredential = New NetworkCredential(strFTPUser, SecurityTools.DecodePassword(EncFTPUserPass))
    Private intSocketTimeout As Integer = 5000

#End Region

#Region "Methods"

    Public Function ReturnFtpRequestStream(uri As String, method As String) As IO.Stream
        Dim request As FtpWebRequest = DirectCast(FtpWebRequest.Create(uri), FtpWebRequest)
        With request
            .Proxy = New WebProxy() 'set proxy to nothing to bypass .NET auto-detect process. This speeds up the initial connection greatly.
            .Credentials = FTPcreds
            .Method = method
            .ReadWriteTimeout = intSocketTimeout
            Return .GetRequestStream
        End With
    End Function

    Public Function ReturnFtpResponse(uri As String, method As String) As WebResponse
        Dim request As FtpWebRequest = DirectCast(FtpWebRequest.Create(uri), FtpWebRequest)
        With request
            .Proxy = New WebProxy() 'set proxy to nothing to bypass .NET auto-detect process. This speeds up the initial connection greatly.
            .Credentials = FTPcreds
            .Method = method
            .ReadWriteTimeout = intSocketTimeout
            Return .GetResponse
        End With
    End Function

#End Region

End Class

File: AssetManager/Data/DataComms/AssetManager/FTP/FTPComms.cs
using System.Net;
using System.IO;

namespace AssetManager
{

    public class FtpComms
    {

        #region "Fields"

        private const string EncFTPUserPass = "BzPOHPXLdGu9CxaHTAEUCXY4Oa5EVM2B/G7O9En28LQ=";
        private const string strFTPUser = "asset_manager";
        private NetworkCredential FTPcreds = new NetworkCredential(strFTPUser, SecurityTools.DecodePassword(EncFTPUserPass));

        private int intSocketTimeout = 5000;
        #endregion

        #region "Methods"

        public Stream ReturnFtpRequestStream(string uri, string method)
        {
            FtpWebRequest request = (FtpWebRequest)FtpWebRequest.Create(uri);
            request.Proxy = new WebProxy();
            //set proxy to nothing to bypass .NET auto-detect process. This speeds up the initial connection greatly.
            request.Credentials = FTPcreds;
            request.Method = method;
            request.ReadWriteTimeout = intSocketTimeout;
            request.Timeout = intSocketTimeout;
            return request.GetRequestStream();
        }

        public WebResponse ReturnFtpResponse(string uri, string method)
        {
            FtpWebRequest request = (FtpWebRequest)FtpWebRequest.Create(uri);
            request.Proxy = new WebProxy();
            //set proxy to nothing to bypass .NET auto-detect process. This speeds up the initial connection greatly.
            request.Credentials = FTPcreds;
            request.Method = method;
            request.ReadWriteTimeout = intSocketTimeout;
            request.Timeout = intSocketTimeout;
            return request.GetResponse();
        }

        #endregion

    }
}
